File: buildSrc/src/main/java/com/unifidokey/gradle/util/VersionCodeUtil.java
package com.unifidokey.gradle.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class VersionCodeUtil {

    public static Integer getVersionCodeFromVersionString(String versionString) {
        Matcher matcher = Pattern.compile("([0-9]+)\\.([0-9]+)\\.([0-9]+)(-SNAPSHOT|\\.RELEASE)").matcher(versionString);
        if (!matcher.find()) {
            throw new IllegalArgumentException("versionString doesn't conform format contract.");
        }
        int major = Integer.parseInt(matcher.group(1));
        int minor = Integer.parseInt(matcher.group(2));
        int patch = Integer.parseInt(matcher.group(3));
        return major * 10000 + minor * 100 + patch;
    }
}

File: buildSrc/src/main/java/com/unifidokey/gradle/util/VersionCodeUtil.kt
package com.unifidokey.gradle.util

import java.util.regex.Pattern

object VersionCodeUtil {
    fun getVersionCodeFromVersionString(versionString: String): Int {
        val matcher = Pattern.compile("([0-9]+)\\.([0-9]+)\\.([0-9]+)(-SNAPSHOT|\\.RELEASE)")
            .matcher(versionString)
        require(matcher.find()) { "versionString doesn't conform format contract." }
        val major = matcher.group(1).toInt()
        val minor = matcher.group(2).toInt()
        val patch = matcher.group(3).toInt()
        return major * 10000 + minor * 100 + patch
    }
}